src/DataProtection/DataProtection/src/Repositories/IDeletableXmlRepository.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.DataProtection.Repositories;
+
+/// <summary>
+/// An extension of <see cref="IXmlRepository"/> that supports deletion of elements.
+/// </summary>
+public interface IDeletableXmlRepository : IXmlRepository
+{
+    /// <summary>
+    /// Removes selected elements from the repository.
+    /// </summary>
+    /// <param name="chooseElements">
+    /// A snapshot of the elements in this repository.
+    /// For each, set <see cref="IDeletableElement.DeletionOrder"/> to a non-<c>null</c> value if it should be deleted.
+    /// Elements are deleted in increasing order.  If any deletion fails, the remaining deletions *MUST* be skipped.
+    /// </param>
+    /// <returns>
+    /// True if all deletions succeeded.
+    /// </returns>
+    bool RemoveElements(Action<IReadOnlyCollection<IDeletableElement>> chooseElements);
```

Comment:
Inconsistent naming Remove vs Delete?

Reply:
Do you remember why we did this, @halter73?  I'm pretty sure it was intentional.  If not, are you okay with changing it to `DeleteElements`, etc?

Reply:
My guess is that `IXmlRepository` is collection-like and collections usually use "Remove".

Reply:
I don't mind it being inconsistent with `IKeyManager`, but I do mind it being inconsistent with `IDeletableXmlRepository`.

Reply:
Renamed to DeleteElements.